src/AzureBlobStorageAdapter.php

```diff
@@ -1,115 +1,350 @@
+        return $this->normalizeProperties($properties, $path);
+    }
+
+    public function fileSize(string $path): FileAttributes
+    {
+        $properties = $this->getMetaData($path);
+
+        return $this->normalizeProperties($properties, $path);
+    }
+
+    public function listContents(string $path, bool $deep): iterable
+    {
+        $dirLocation = $this->prefixer->prefixDirectoryPath($path);
+        $options = new ListBlobsOptions();
+        $options->setPrefix($dirLocation);
+        $listResults = $this->client->listBlobs($this->container, $options);
+
+        return $listResults->getBlobs();
+    }
+
+    public function move(string $source, string $destination, Config $config): void
+    {
+        $this->copy($source, $destination, $config);
+        $this->delete($source);
+    }
+
+    public function copy(string $source, string $destination, Config $config): void
+    {
+        $this->initialize();
+        $this->logger->debug("copy from the location {$source}");
+        try {
+            $sourceLocation = $this->prefixer->prefixPath($source);
+            $destinationLocation = $this->prefixer->prefixPath($destination);
+            $this->client->copyBlob(
+                $this->container,
+                $destinationLocation,
+                $this->container,
+                $sourceLocation
+            );
+        } catch (ServiceException $exception) {
+            $this->logger->error("Error from the location {$source}");
+            throw new UnableToCopyFile("Unable to find the file {$source}");
+        }
+    }
+
+    private function getMetaData(string $path): GetBlobPropertiesResult
+    {
+        $this->initialize();
+
+        $location = $this->prefixer->prefixPath($path);
+
+        return $this->client->getBlobProperties(
+                $this->container,
+                $location
+            );
+    }
+
+    private function normalizeProperties(
+        GetBlobPropertiesResult $blobPropertiesResult,
+        string $path
+    ): FileAttributes {
+        $properties = $blobPropertiesResult->getProperties();
+
+        return new FileAttributes(
+            $path,
+            $properties->getContentLength(),
+        null,
+            (int) $properties->getLastModified()->format('U'),
+            $properties->getContentType(), [
+
+            ]
+        );
+    }
+
+    private function createOptionsFromConfig(Config $config): CreateBlockBlobOptions
+    {
+        $options = $config->get('blobOptions', new CreateBlockBlobOptions());
+        foreach (static::$metaOptions as $option) {
+            if (!$config->get($option)) {
+                continue;
+            }
+
+            call_user_func([$options, "set$option"], $config->get($option));
+        }
+
+        if ($mimetype = $config->get('mimetype')) {
+            $options->setContentType($mimetype);
+        }
+
+        return $options;
+    }
+
+    private function initialize():void
+    {
+        if (!$this->booted) {
```

Comment:
when did we set the booted property? I didn't see it in the createContainer method definition